frontend/packages/dev-console/src/components/pipelines/modals/triggers/resource-utils.ts

```diff
@@ -12,14 +12,15 @@ import {
 } from '../../resource-types';
 
 export const createTriggerTemplate = (
+  pipeline: Pipeline,
   pipelineRun: PipelineRun,
   params: TriggerTemplateKindParam[],
 ): TriggerTemplateKind => {
   return {
```

Comment:
I wonder if we could just override it for this one use-case rather than baking it into the (more or less) broken flow that exists today getting a Pipeline Run.

I think the refactor portion of your PR hit the nose of the problem, and that definitely is the better choice once we can refactor the code. Just not sure we need to pollute "crafting a pipeline run" if we are going to rewrite it in a few anyways.

I just suggest the modification here: 
```
  const resource = omit(pipelineRun, 'metadata.name');
  resource.metadata.generateName = `${pipeline.metadata.name}-`;
```

Reply:
@andrewballantyne I though multiple times (and longer than I should) about this.. I would be happy if we stay with this (reduced) change. Keep the additional options parameter allow us to extend the unit tests for this case. Keep the additional tests instead of just "override" the generated json here is worth enough to touch 3 files.

I'm on your side that its still not perfect, but for me it looks like the right direction. :thinking: 

I'm also happy if we want split this method into 3 when starting with 4.7.